src/server/lock_manager/config.rs

```diff
@@ -30,9 +37,51 @@ impl Config {
     }
 }
 
+pub struct LockManagerConfigManager {
+    waiter_mgr_scheduler: WaiterMgrScheduler,
+    detector_scheduler: DeadlockScheduler,
+}
+
+impl LockManagerConfigManager {
+    pub fn new(
+        waiter_mgr_scheduler: WaiterMgrScheduler,
+        detector_scheduler: DeadlockScheduler,
+    ) -> Self {
+        LockManagerConfigManager {
+            waiter_mgr_scheduler,
+            detector_scheduler,
+        }
+    }
+}
+
+impl ConfigManager for LockManagerConfigManager {
+    fn dispatch(&mut self, mut change: ConfigChange) -> Result<(), Box<dyn Error>> {
+        match (
+            change.remove("wait_for_lock_timeout").map(Into::into),
+            change.remove("wake_up_delay_duration").map(Into::into),
+        ) {
+            (timeout @ Some(_), delay) => {
```

Comment:
What if `delay` is None?

Reply:
`WaiterManager` need to listening both update of `timeout` and `delay`, even one of them are not change (represent as `None` here), we still need to dispatch the change to it, and `WaiterManager::handle_config_change` will take care of various situations.